src/config/sfdxDataHandler.ts

```diff
@@ -192,3 +196,43 @@ export class AuthHandler extends BaseHandler<SfInfoKeys.ORGS> {
     return auths;
   }
 }
+
+export class AliasesHandler extends BaseHandler<SfInfoKeys.ALIASES> {
+  private static SFDX_ALIASES_FILENAME = 'alias.json';
+
+  public sfKey: typeof SfInfoKeys.ALIASES = SfInfoKeys.ALIASES;
+
+  public async migrate(): Promise<Pick<SfInfo, SfInfoKeys.ALIASES>> {
+    const aliasesFilePath = join(Global.SFDX_DIR, AliasesHandler.SFDX_ALIASES_FILENAME);
+    try {
+      const sfdxAliases = ((await fs.readJson(aliasesFilePath)) as Record<'orgs', Record<string, string>>).orgs;
+      return { [this.sfKey]: { ...sfdxAliases } };
+    } catch (e) {
+      return { [this.sfKey]: {} };
+    }
+  }
+
+  // AliasesHandler implements its own merge method because the structure of aliases is flat instead of nested by SfInfoKey types.
+  public async merge(sfData: SfInfo = GlobalInfo.emptyDataModel): Promise<Partial<SfInfo>> {
+    const sfdxAliases: Record<string, string> = (await this.migrate())[SfInfoKeys.ALIASES];
+    const merged = deepCopy<SfInfo>(sfData);
+
+    // See the merge method of the BaseHandler class for the reasoning behind the two following loops
+    // Overwrite sfAliases with sfdxAliases
+    Object.keys(sfdxAliases).forEach((alias) => {
+      merged[SfInfoKeys.ALIASES][alias] = sfdxAliases[alias];
+    });
+
+    // Delete any keys that are not in sfdxData
```

Comment:
We should add a note here about our assumptions. Specifically, that we can assume sfdx will always have sf's aliases since core v3 always writes those changes back to sfdx

Reply:
Yeah, I added https://github.com/forcedotcom/sfdx-core/pull/444/files/9ccf038aadb2879530b1bdbb09428d4e6f5258cd#diff-36c95a1919f4451b1f5aca8a051b155264af593e995ef4555b885eaededbfb22R220 to help with that but kinda hard to see it applies to this loop too.